src/OneSpanSign.ts

```diff
@@ -0,0 +1,39 @@
+import FormData from 'form-data';
+
+import { Api } from './api';
+import {
+  CreatePackageRequestPayload,
+  CreatePackageResponsePayload,
+  DocumentMetadata,
+  UploadDocumentRequestPayload,
+} from './types';
+
+export class OneSpanSign {
+  constructor(private apiKey: string, private apiUrl: string) {}
+
+  public async createPackage(payload: CreatePackageRequestPayload): Promise<CreatePackageResponsePayload> {
+    const response = await Api.post(`${this.apiUrl}/api/packages`)
+      .withAuthorizationHeader(`Basic ${this.apiKey}`)
+      .withBody(payload)
+      .fetch();
+
+    return (await response.json()) as CreatePackageResponsePayload;
+  }
+
+  public async uploadDocument(
+    packageId: string,
+    payload: UploadDocumentRequestPayload,
+    documentBody: Buffer | ReadableStream
```

Comment:
Using `ReadableStream`, we could pipe the download stream from cloud storage servers (e.g. [Google Cloud Storage](https://cloud.google.com/storage/docs/streaming)) directly to the form data object

Reply:
Nice 👍